Framework/DataHandling/src/LoadInstrument.cpp

```diff
@@ -150,8 +152,21 @@ void LoadInstrument::exec() {
     const std::string::size_type stripPath = filename.find_last_of("\\/");
     std::string instrumentFile =
         filename.substr(stripPath + 1, filename.size());
+
     // Strip off "_Definition.xml"
-    instname = instrumentFile.substr(0, instrumentFile.find("_Def"));
+    auto definitionRange = boost::ifind_first(instrumentFile, "_Def");
+    if (definitionRange) {
+      // We need to find the position this occurs in the string and we
+      // can't pass begin() to substr. So take a performance pessimism
+      // and create a copy which find is guaranteed to find
+      const std::string foundDefTagCase(definitionRange.begin(),
+                                        definitionRange.end());
+      instname = instrumentFile.substr(0, instrumentFile.find(foundDefTagCase));
+    } else {
+      g_log.warning("The instrument definition filename does not contain "
```

Comment:
Should be more explicit in the warning and specify that we are setting the instrument name to `instrumentFile`?